data/src/main/java/org/apache/iceberg/data/DeleteFilter.java

```diff
@@ -243,20 +236,9 @@ private CloseableIterable<T> applyPosDeletes(CloseableIterable<T> records) {
       return records;
     }
 
-    List<CloseableIterable<Record>> deletes = Lists.transform(posDeletes, this::openPosDeletes);
-
-    // if there are fewer deletes than a reasonable number to keep in memory, use a set
-    if (posDeletes.stream().mapToLong(DeleteFile::recordCount).sum() < setFilterThreshold) {
-      PositionDeleteIndex positionIndex = Deletes.toPositionIndex(filePath, deletes);
-      Predicate<T> isDeleted = record -> positionIndex.isDeleted(pos(record));
-      return createDeleteIterable(records, isDeleted);
-    }
-
-    return hasIsDeletedColumn
```

Comment:
This is a separate discussion but I think we should drop streaming position deletes. We originally added them when we used a set, not bitmaps.